apidef/oas/middleware.go

```diff
@@ -720,3 +737,34 @@ func (et *EnforceTimeout) ExtractTo(meta *apidef.HardTimeoutMeta) {
 	meta.Disabled = !et.Enabled
 	meta.TimeOut = et.Value
 }
+
+type CustomPluginMiddleware struct {
+	// FunctionName is the name of authentication method.
+	FunctionName string `bson:"functionName,omitempty" json:"functionName,omitempty"`
+	// Path is the path to shared object file in case of gopluign mode or path to js code in case of otto auth plugin.
+	Path string `bson:"path,omitempty" json:"path,omitempty"`
+	// RawBodyOnly if set to true, do not fill body in request or response object.
+	RawBodyOnly bool `bson:"rawBodyOnly,omitempty" json:"rawBodyOnly,omitempty"`
+}
+
+// AuthenticationPlugin holds the configuration for custom authentication plugin.
+type AuthenticationPlugin struct {
+	// Enabled enables custom authentication plugin.
+	Enabled bool `bson:"enabled,omitempty" json:"enabled,omitempty"` // required.
+	// CustomPluginMiddleware holds configuration for custom middleware.
+	CustomPluginMiddleware `bson:",inline" json:",inline"`
```

Comment:
Why is this inlined? If it is used multiple times, why don't it include `enabled inside` ?

Reply:
We can use `CustomPluginMiddleware` for other middlewares